### PR TITLE
Replaced merge-graphql-schemas with graphql-tools

### LINKED ISSUE #1230: Replace merge-graphql-schemas with graphql-tools
We're using `merge-graphql-schemas` which is now deprecated: https://github.com/redwoodjs/redwood/blob/e5574f9eba1578f1dc03ba30fa374e6a5083de3a/packages/api/src/makeMergedSchema/makeMergedSchema.ts#L7

Let's rather use graphql-tools/merge package:
https://www.graphql-tools.com/docs/migration-from-merge-graphql-schemas/

I think this may be required for directives to work properly.

### PR DESCRIPTION
**Pull request by himankpathak**
### Description
Replaced `mergeType` from merge-graphql-schemas with `mergeTypeDefs` graphql-tools

### Verification Process
Tested on local with prebuilt tests

### Applicable Issues
Closes #1230

### CODE REVIEW

**Comment by thedavidprice** [comment]
Awesome, thanks for getting this rolling @himankpathak 🚀 

Handing you off to @peterp from here out.

**Comment by peterp** [comment]
@himankpathak I'm surprised that the tests are working locally since this doesn't seem to be building properly. Perhaps I over estimated the coverage that we have.

**Comment by himankpathak** [comment]
@peterp, I am surprised too. Locally test passed. But on PR the build is failing.

File: packages/api/src/makeMergedSchema/makeMergedSchema.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ export const makeMergedSchema = ({
   services: Services
   schemaDirectives?: IExecutableSchemaDefinition['schemaDirectives']
 }) => {
-  const typeDefs = mergeTypes(
+  const typeDefs = mergeTypeDefs(
     [rootSchema.schema, ...Object.values(schemas).map(({ schema }) => schema)],
```
Comment (himankpathak) [inline packages/api/src/makeMergedSchema/makeMergedSchema.ts:159]: @peterp, We get `No overload matches this call.` from this line.
Wrapping it up in lodash.merge fixes the error for this line.

File: packages/api/src/makeMergedSchema/makeMergedSchema.ts [hunk 2]
```diff
@@ -155,7 +155,7 @@ export const makeMergedSchema = ({
...
     { all: true }
```
Comment (himankpathak) [inline packages/api/src/makeMergedSchema/makeMergedSchema.ts:160]: Fixing the error in the line above gives us same error in the following line `No overload matches this call.`

**Comment by Urigo** [comment]
Just a small note - if you want to use the new GraphQL Tools, you better install the scoped packages, so you'll get only the parts of `graphql-tools` that you need

**Comment by peterp** [comment]
Hey @himankpathak 

It looks like it's the same issue as before - I'll check out your branch in a few minutes and figure this out.

**Comment by himankpathak** [comment]
That would be great! @peterp

**Comment by thedavidprice** [comment]
@peterp is this PR + work still relevant. If so, @himankpathak are you still interested in working on this?

Thanks to you both!

**Comment by himankpathak** [comment]
@thedavidprice definitely!
I will resolve the conflicts and see if we sort the failing tests.

**Comment by peterp** [comment]
Thanks @himankpathak!

**Comment by himankpathak** [comment]
Test cases are passing on local

We have following error on `packages/api/src/makeMergedSchema/makeMergedSchema.ts` L167
```
No overload matches this call.
  Overload 1 of 3, '(types: (string | GraphQLSchema | Source | DocumentNode)[], config?: (Partial<Config> & { commentDescriptions: true; }) | undefined): string', gave the following error.
    Type 'Record<string, unknown>' is not assignable to type 'string | GraphQLSchema | Source | DocumentNode'.
      Type 'Record<string, unknown>' is not assignable to type 'string'.
  Overload 2 of 3, '(types: (string | GraphQLSchema | Source | DocumentNode)[], config?: Pick<Partial<Config>, "useSchemaDefinition" | "forceSchemaDefinition" | ... 4 more ... | "convertExtensions"> | undefined): DocumentNode', gave the following error.
    Type 'Record<string, unknown>' is not assignable to type 'string | GraphQLSchema | Source | DocumentNode'.
      Type 'Record<string, unknown>' is not assignable to type 'string'.ts(2769)
```

**Comment by thedavidprice** [comment]
@peterp I just pushed code from the [Migration from Merge GraphQL Schemas](https://www.graphql-tools.com/docs/migration-from-merge-graphql-schemas/) guide that proxies `mergeTypes` to `mergeTypeDefs`. There was an API change — this solution was an, ahem, efficient step forward although I'm sure it could be condensed.

Tested on my local project and everything is working fine (note the need to account for [this change](https://github.com/redwoodjs/redwood/pull/1639#issuecomment-767166900) to test a project). Tests passed locally.

**Review by peterp — APPROVED** [review]
Looks good to me!

**Comment by peterp** [comment]
Before we release v0.24.0 I would like to try this out in Snaplet.

**Pull request merged**